### PR TITLE
Support MDS in feature anywhere

### PR DESCRIPTION
**Pull request by riysaxen-amzn**
### Description
* This PR adds multi data source support to Feature-Anywhere from Alerting side
* This PR also resolves some Console errors
 
### Issues Resolved

[965](https://github.com/opensearch-project/alerting-dashboards-plugin/issues/965)

### Testing

https://github.com/opensearch-project/alerting-dashboards-plugin/assets/69919272/78fd6a1c-4da6-4d05-8f91-e44b52bb15e4

Associated Monitors

https://github.com/opensearch-project/alerting-dashboards-plugin/assets/69919272/d2681e39-61fb-4d7a-b6b5-8bfe0ea55ed0

 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/alerting-dashboards-plugin/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: public/utils/contextMenu/actions.tsx [hunk 1]
```diff
@@ -38,6 +44,8 @@ export const openContainerInFlyout = async ({
   embeddable: any;
   detectorId?: string;
 }) => {
+  const indexPatternId = embeddable.vis.data.aggs.indexPattern.id;
```
Comment (BionIT) [inline public/utils/contextMenu/actions.tsx:47]: I think you should be able to get data source id directly from the embeddable object `embeddable.vis.data.indexPattern.dataSourceRef.id`

Comment (riysaxen-amzn) [reply public/utils/contextMenu/actions.tsx]: changed now

File: public/pages/CreateTrigger/containers/ConfigureTriggers/ConfigureTriggers.js [hunk 2]
```diff
@@ -217,7 +217,7 @@ class ConfigureTriggers extends React.Component {
   }
 
   async onQueryMappings() {
-    const indices = this.props.monitor.inputs[0].search.indices;
+    const indices = this.props.monitor.inputs[0].search?.indices;
```
Comment (AWSHurneyt) [inline public/pages/CreateTrigger/containers/ConfigureTriggers/ConfigureTriggers.js:220]: Quick clarification question. Do we need this to default to an empty list in situations where it's undefined?
E.g.,
```
const indices = this.props.monitor.inputs[0].search?.indices || [];

File: public/pages/CreateTrigger/containers/CreateTrigger/CreateTrigger/CreateTrigger.js [hunk 3]
```diff
@@ -282,7 +282,7 @@ export default class CreateTrigger extends Component {
   }
 
   async onQueryMappings() {
-    const indices = this.props.monitor.inputs[0].search.indices;
+    const indices = this.props.monitor.inputs[0].search?.indices;
```
Comment (AWSHurneyt) [inline public/pages/CreateTrigger/containers/CreateTrigger/CreateTrigger/CreateTrigger.js:285]: Same question as above.

File: public/utils/contextMenu/monitors.ts [hunk 4]
```diff
@@ -34,7 +34,7 @@ const parseMonitor = (monitor) => {
     date: latestAlert, // this is the last alert time
     id: monitor.id,
     type: monitor.monitor.monitor_type,
-    indexes: monitor.monitor.inputs[0].search.indices,
+    indexes: monitor.monitor.inputs[0].search?.indices,
```
Comment (AWSHurneyt) [inline public/utils/contextMenu/monitors.ts:37]: Same question as above.

File: public/pages/CreateTrigger/containers/ConfigureTriggers/ConfigureTriggers.js (hunk 2 above)
Comment (riysaxen-amzn) [reply public/pages/CreateTrigger/containers/ConfigureTriggers/ConfigureTriggers.js]: addressed

**Review by AWSHurneyt — APPROVED** [review]

File: public/utils/contextMenu/actions.tsx [hunk 5]
```diff
@@ -118,3 +125,14 @@ export const getAdAction = () =>
         defaultFlyoutMode: 'adMonitor',
       }),
   });
+
+function setDataSourceIdFromSavedObject(indexPattern: any) {
```
Comment (amsiglan) [inline public/utils/contextMenu/actions.tsx:129]: Don't use `any` if possible

Comment (amsiglan) [reply public/utils/contextMenu/actions.tsx:129]: Will change in follow up PR

**Review by amsiglan — APPROVED** [review]

**Pull request merged**